Drop Kanban development process, rework common labels (#7880)
nextcloud/documentation: developer_manual/getting_started/codingguidelines.rst

The most important labels and their meaning:

* Tags showing the state of the issue or PR, numbered 0-4:

>   * ``0. to triage`` - ready to start development on this

Reviewer: ```suggestion
  * ``0. to triage`` - issue or feature request needs to get triaged and approved for development
```

Open to other suggestions on the description but should be different from "1. to develop"

  * ``1. to develop`` - ready to start development on this
  * ``2. developing`` - development in progress
  * ``3. to review`` - ready for review

>   * ``4. to release`` - reviewed PR that awaits unfreeze of a branch to get merged

Reviewer: ```suggestion
  * ``4. to release`` - reviewed PR that awaits unfreeze of a branch to get merged or has pending CI jobs
```

  * ``needs info`` - this issue needs further information from the reporter, see :doc:`../../prologue/bugtracker/triaging`. This tag is typically combined with ``0. to triage`` to signal a bug report is not confirmed yet or a feature request has not been approved.

* Tags showing the type of issue or PR

  * ``bug`` - this issue is a bug
  * ``enhancement`` - this issue is a feature request/idea for improvement of Nextcloud
  * ``technical debt`` - this issue or PR is about `technical debt <https://en.wikipedia.org/wiki/Technical_debt>`_

* Tags that classify an issue or PR

>   * ``high``, ``medium`` ``low`` signify how important the bug is.

Reviewer 2: ```suggestion
  * ``high``, ``medium``, ``low`` signify how important the bug is.
```

>   * ``high``, ``medium`` and ``low`` signify how important the bug is.

Reviewer 2: ```suggestion
  * ``high``, ``medium`` and ``low`` - signify how important the bug is
```
To unify the list?

  * ``regression`` - something that worked in a previous release but is now not working as expected or missing.

>   * ``feature: *``, e.g. ``feature: dav``. These tags group tickets of specific feature or subsystems.

Reviewer 2: ```suggestion
  * ``feature: *``, e.g. ``feature: dav`` - these tags group tickets of specific feature or subsystems
```
Maybe?

  * ``design`` - this needs help from the design team or is a design-related issue/pull request
  * ``good first issue`` - these are issues which are relatively easy to solve and ideal for people who want to learn how to code in Nextcloud

> * ``backport-request`` - the pull requests also needs to be applied to older Nextcloud versions. This tag is typically assigned by automation.

Reviewer 2: Is the automation / bot documented somewhere? Maybe a link or a sentence or two on how it works?

Author: As so many things: no, it's not documented at all, so I raised https://github.com/nextcloud/documentation/issues/7881

> If you want a label not in the list above, please first discuss on the mailing list.

Reviewer 2: Does not make sense anymore (https://github.com/nextcloud/documentation/pull/7879)?
